### PR TITLE
Feature/tdp 2704 side panel integration

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2704

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)

**Please check the browsers you've tested on**

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge
- [x] IE11

**(Optional) What is the current behavior?**
IE11: KO because of a problem from talend-icon

### CODE REVIEW

File: dataprep-webapp/src/app/components/widgets-containers/app-layout/app-layout-container.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ This source code is available under agreement available at
+ https://github.com/Talend/data-prep/blob/master/LICENSE
+
+ You should have received a copy of the agreement
+ along with this program; if not, write to Talend SA
+ 9 rue Pages 92150 Suresnes, France
+
+ ============================================================================*/
+const AppLayoutContainer = {
+	template:  `<div class="app">
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/app-layout/app-layout-container.js]: Why did you not use `Layout` component from react-talend-components instead?

Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/widgets-containers/app-layout/app-layout-container.js]: In order to use Layout component from react-talend-components, you have to use the react "sidePanel" component as a prop. The problem is you can not use a react component with Angular.

So I have to create an Angular Layout component whose template looks like the one of  Layout component from react-talend-components

File: dataprep-webapp/src/app/components/widgets-containers/app-layout/app-layout.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    .sidemenu {
+      @include flex-grow(0);
+      @include flex-shrink(1);
+
+      background-color: $blue;
+
+      //HACK: to remove after PR react-talend-components
+      .fa-arrow-left:before {
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/app-layout/app-layout.scss]: Arrow doesn't rotate when it's docked :(

File: dataprep-webapp/src/app/components/widgets-containers/app-side-panel/app-side-panel-container.spec.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	}));
+
+	afterEach(inject((SettingsService) => {
+		SettingsService.clearSettings();
+		scope.$destroy();
+		element.remove();
+	}));
+
+	describe('render', () => {
+		it('should create toogle button', () => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/app-side-panel/app-side-panel-container.spec.js]: tOOgle spotted!

File: dataprep-webapp/src/app/services/state/home/home-state-service.spec.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	beforeEach(angular.mock.module('data-prep.services.state'));
+
+
+	it('should toogle sidepanel', inject(function (homeState, HomeStateService) {
+		//given
+		homeState.sidePanelDocked = false;
+
+		//when
+		HomeStateService.toogleSidepanel();
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/state/home/home-state-service.spec.js]: toogle

**Pull request merged**